core/src/main/java/com/novoda/rxpresso/mock/RxMocks.java

```diff
@@ -0,0 +1,240 @@
+    private final Map<Observable, Pair<ClearableBehaviorSubject<Notification>, PublishSubject<Notification>>> mapSubject = new HashMap<>();
+
+    public static RxMocks mock(Class clazz) {
+        return init(Mockito.mock(clazz));
+    }
+
+    public static RxMocks init(Object repository) {
+        RxMocks rxMocks = new RxMocks(repository);
+        rxMocks.setMockResponses();
+        return rxMocks;
+    }
+
+    private RxMocks(Object repository) {
+        this.repository = repository;
+    }
+
+    private void setMockResponses() {
+        for (Method method : repository.getClass().getMethods()) {
+            if (method.getReturnType().equals(Observable.class) && isMockable(method)) {
+                setupMockResponseFor(method);
+            }
+        }
+    }
+
+    private boolean isMockable(Method method) {
+        return !Modifier.isPrivate(method.getModifiers())
+                && !Modifier.isProtected(method.getModifiers())
+                && !Modifier.isStatic(method.getModifiers());
+    }
+
+    public <T> Boolean provides(Observable<T> observable) {
+        return observableHashMap.containsValue(observable);
+    }
+
+    public <T> Observable<Notification<T>> getEventsFor(Observable<T> observable) {
+        Pair<ClearableBehaviorSubject<Notification>, PublishSubject<Notification>> subjectPair = mapSubject.get(observable);
+        if (subjectPair == null) {
+            throw new IllegalArgumentException(
+                    "The observable " + observable +
+                            " is not provided by this repo use the provides(Observable o) method to check first");
+        }
+        return Observable.zip(subjectPair.first, subjectPair.second, unzip())
+                .lift(clearOnUnsubscribe(observable));
+    }
+
+    /**
+     * Inject the events from given {@code source} into a mocked pipeline
+     *
+     * @param source The observable producing the events to inject
+     * @param <T>    The type of this observable
+     * @return A sender object to define into which pipeline to inject the events.
+     */
+    public <T> RxObservableSender<T> sendEventsFrom(Observable<T> source) {
+        return new RxObservableSender<>(source);
+    }
+
+    public class RxObservableSender<T> {
+
+        private final Observable<T> source;
+
+        public RxObservableSender(Observable<T> source) {
+            this.source = source;
+        }
+
+        /**
+         * Send the events from {@code source} to the given mocked {@code observable}
+         *
+         * @param observable The mocked observable to inject events into.
+         */
+        public void to(Observable<T> observable) {
+            ((Observable) source).materialize().lift(infinite()).subscribe(mapSubject.get(observable).first);
+        }
+
+    }
+
+    public void resetMocks() {
+        observableHashMap.clear();
+        mapSubject.clear();
+    }
+
+    private void setupMockResponseFor(Method method) {
```

Comment:
Can you reduce the scope of the try catch by extracting a method handle that part?

```java
private void setupMockResponseFor(Method method) {
    when(invoking(method)).thenAnswer(
            new Answer<Observable>() {
                @Override
                public Observable answer(InvocationOnMock invocation) throws Throwable {
                    Method method = invocation.getMethod();
                    Object[] arguments = invocation.getArguments();

                    String key = getKeyFor(method, arguments);
                    if (!observableHashMap.containsKey(key)) {
                        initialiseMockedObservable(method, arguments);
                    }

                    return observableHashMap.get(key);
                }
            }
    );
}

private Object invoking(Method method) {
    try {
        return method.invoke(repository, getArgumentsFor(method));
    } catch (IllegalAccessException e) {
        e.printStackTrace();
    } catch (InvocationTargetException e) {
        e.printStackTrace();
    }
    return null;
}
```

The return null at the bottom will probably fail in some unknown way (`when(null)`?) but if we're not sure what to do here, and don't know how to react to the caught exceptions, should we not explode with a runtime error?